### PR TITLE
Consolidate the GetDiagnosticInfo DM handling

### PR DESCRIPTION
**Pull request by cristipogacean**
* do not ignore default values for the various metrics
 * cosmetic changes
 * alignment in naming for endpoint in the reply payload

### CODE REVIEW

**Comment by dacolgit** [comment]
> //  Copyright (c) Microsoft Corporation.  All rights reserved.

I don't see any changes in this file

---
In reply to: [1050058477](https://github.com/Azure/Industrial-IoT/pull/1596#issuecomment-1050058477) [](http://example.com/codeflow?ancestors=1050058477)

---
Refers to: common/src/Microsoft.Azure.IIoT.Agent.Framework/src/Models/AuthMode.cs:2 in 9d484e2. [](commit_id = 9d484e20c54ffd078abfc6342c0b4b860337a31b, deletion_comment = False)

**Review by dacolgit — APPROVED** [review]
:shipit:

File: api/src/Microsoft.Azure.IIoT.OpcUa.Api.Publisher/src/Models/DiagnosticInfoApiModel.cs [hunk 1]
```diff
@@ -4,128 +4,128 @@
...
     public class DiagnosticInfoApiModel {
- 
+
         /// <summary> EndpointInfo /// </summary>
-        [DataMember(Name = "EndpointInfo", Order = 0,
-            EmitDefaultValue = false)]
-        public PublishNodesEndpointApiModel EndpointInfo { get; set; }
+        [DataMember(Name = "endpoint", Order = 0,
+            EmitDefaultValue = true)]
+        public PublishNodesEndpointApiModel Endpoint { get; set; }
```
Comment (karok2m) [inline api/src/Microsoft.Azure.IIoT.OpcUa.Api.Publisher/src/Models/DiagnosticInfoApiModel.cs:19]: Let's update the documentation accordingly once #1582 is merged.

**Comment by cristipogacean** [comment]
> //  Copyright (c) Microsoft Corporation.  All rights reserved.

just some trailing spaces after \<summary\> and the explanation 

---
In reply to: [1050058477](https://github.com/Azure/Industrial-IoT/pull/1596#issuecomment-1050058477) [](http://example.com/codeflow?ancestors=1050058477,1050058477)

---
Refers to: common/src/Microsoft.Azure.IIoT.Agent.Framework/src/Models/AuthMode.cs:2 in 9d484e2. [](commit_id = 9d484e20c54ffd078abfc6342c0b4b860337a31b, deletion_comment = False)

File: api/src/Microsoft.Azure.IIoT.OpcUa.Api.Publisher/src/Models/DiagnosticInfoApiModel.cs (hunk 1 above)
Comment (cristipogacean) [reply api/src/Microsoft.Azure.IIoT.OpcUa.Api.Publisher/src/Models/DiagnosticInfoApiModel.cs:19]: absolutely. will do in a separate PR after https://github.com/Azure/Industrial-IoT/pull/1582 is merged

Comment (monikavar) [reply api/src/Microsoft.Azure.IIoT.OpcUa.Api.Publisher/src/Models/DiagnosticInfoApiModel.cs:19]: PR is merged now.

**Review by karok2m — APPROVED** [review]

**Review by monikavar — APPROVED** [review]

**Pull request merged**